src/Features/CSharpTest/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -8984,4 +8985,303 @@ public class C
             }
             """);
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67270")]
+    public async Task TestTopLevel1()
+    {
+        await TestAsync(
+            """
+            using System;
+
+            var v1 = [|new Random()|].Next();
+            var v2 = new Random().Next();
+            """,
+            """
+            using System;
+
+            var {|Rename:random|} = new Random();
+
+            var v1 = random.Next();
+            var v2 = random.Next();
```

Comment:
>random [](http://example.com/codeflow?start=21&length=6)

Haven't looked at the changes, but did it make it so the "new Random()" call is only done once?

Reply:
yes.  but this is testing "replace all occurrences of this pattern"